flang/include/flang/Optimizer/Dialect/FIRTypes.td

```diff
@@ -213,6 +213,22 @@ def fir_IntegerType : FIR_Type<"Integer", "int"> {
   }];
 }
 
+def fir_UnsignedType : FIR_Type<"Unsigned", "unsigned"> {
```

Comment:
Did you actually need this new type?

fir_IntegerType still exits but likely has very limited use in lowering, it is my bad for not removing it completely. So I would not add this new type if there is no use for it (it looked properly plugged in otherwise).

Reply:
I get HLFIR verifier errors on things like unsigned assignments if that type isn't defined and included in the sets of types below in that file.

Reply:
I understand the `AnyUnsignedInteger.predicate` is needed, but I do not understand why the predicate for this type is needed given I do not see where it is generated/used (which I may just have missed in the patch).

Reply:
Removing both the type and predicate, I got no failures (I had to help make by deleting some of the tablegen generated `.cpp.inc` that still referred to the type though, but fresh build should not hit that issue). I am however OK with adding this type for consistency with fir_IntegerType.